core/src/androidTest/java/com/facebook/testing/screenshot/internal/ScreenshotImplTest.java

```diff
@@ -125,6 +131,32 @@ public void testRecordBuilderImplHasAHierarchyDumpFile() throws Throwable {
     MoreAsserts.assertContainsRegex("blahblah.*.xml", metadataContents);
   }
 
+  @Test
+  public void testRecordBuilderImplSkipsHierarchyDumpFile() throws Throwable {
+    Bundle noDumpArgs = new Bundle();
+    noDumpArgs.putString("no_dump", "true");
+
+    Registry registry = Registry.getRegistry();
+    Bundle oldArguments = registry.arguments;
+    registry.arguments = noDumpArgs;
+    try {
+      mScreenshot = ScreenshotImpl.getInstance();
```

Comment:
new ScreenshotImpl(...)

Since you're testing ScreenshotImpl you don't want to use the singleton version which is used in real code.